### PR TITLE
Move GetMarkets definition to generic langchain.tools representation

### PR DESCRIPTION
**Pull request by evangriffiths**
Picking langchain tools as the generic representation https://python.langchain.com/v0.1/docs/modules/tools/, because:
- they are useable directly by crewai
- they're easily convertible to openai tool schema via https://api.python.langchain.com/en/latest/utils/langchain_core.utils.function_calling.convert_to_openai_function.html 
  - This is one way to register functions in autogen -- see https://github.com/microsoft/autogen/blob/main/notebook/agentchat_langchain.ipynb
  - and is also good if we're cooking our own agent framework
  
Fixes https://github.com/gnosis/prediction-market-agent/issues/237

### CODE REVIEW

File: prediction_market_agent/agents/microchain_agent/market_functions.py [hunk 1]
```diff
@@ -311,16 +295,25 @@ def __call__(self) -> list[str]:
...
-    BuyYes,
-    BuyNo,
-    SellYes,
-    SellNo,
-    GetLiquidPositions,
-]
+def build_market_functions(market_type: MarketType) -> list[MarketFunction]:
+    # Functions that interact with the prediction markets
+    market_functions: list[type[MarketFunction]] = [
+        GetMarketProbability,
```
Comment (evangriffiths) [inline prediction_market_agent/agents/microchain_agent/market_functions.py:301]: The rest will all get moved to `prediction_market_agent.tools.prediction_market` too - this is just a to show an example of what moving one looks like

File: prediction_market_agent/agents/microchain_agent/functions_from_tools.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from langchain.tools import BaseTool
+from microchain import Function
+
+
+def microchain_function_from_tool(tool: BaseTool, example_args: list[str]) -> Function:
```
Comment (evangriffiths) [inline prediction_market_agent/agents/microchain_agent/functions_from_tools.py:5]: `example_args` isn't required for langchain/openai tools, but is for microchain Functions

File: prediction_market_agent/tools/prediction_market.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# from typing import Type
+
+# from langchain.pydantic_v1 import BaseModel, Field
```
Comment (gabrielfior) [inline prediction_market_agent/tools/prediction_market.py:3]:
```suggestion
```

**Review by gabrielfior — APPROVED** [review]

File: prediction_market_agent/tools/prediction_market.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# from typing import Type
```
Comment (kongzii) [inline prediction_market_agent/tools/prediction_market.py:1]:
```suggestion
```

File: prediction_market_agent/tools/prediction_market.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class GetMarkets(MarketTool):
+    name: str = "get_markets"
+    description: str = (
+        "Use this function to get a list of predction market questions, and "
+        "the corresponding market IDs"
+    )
+
+    # class Input(BaseModel):
```
Comment (kongzii) [inline prediction_market_agent/tools/prediction_market.py:29]:
```suggestion
```

File: prediction_market_agent/tools/prediction_market.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    #     foo: str = Field(description="bar")
```
Comment (kongzii) [inline prediction_market_agent/tools/prediction_market.py:30]:
```suggestion
```

File: prediction_market_agent/tools/prediction_market.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    # args_schema: Type[BaseModel] = Input
```
Comment (kongzii) [inline prediction_market_agent/tools/prediction_market.py:31]:
```suggestion
```

File: tests/agents/microchain/test_functions.py [hunk 8]
```diff
@@ -196,3 +192,34 @@ def test_remember_past_learnings(long_term_memory: LongTermMemory) -> None:
...
     print(remember_past_learnings())
+
+
+def test_build_market_functions() -> None:
+    functions = build_market_functions(market_type=MarketType.OMEN)
+    assert len(functions) > 0
+
+
+def test_microchain_function_from_tool() -> None:
+    # @tool
```
Comment (kongzii) [inline tests/agents/microchain/test_functions.py:203]:
```suggestion
```

**Review by kongzii — APPROVED** [review]
some forgotten comments otherwise lgtm

**Comment by evangriffiths** [comment]
closing for now - see comments on issue

**Pull request closed without merging**